extensions/jdbc/jdbc-mssql/jdbc-mssql-deployment/src/main/java/io/quarkus/jdbc/mssql/deployment/MsSQLReflections.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.jdbc.mssql.deployment;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
+
+/**
+ * Registers the {@code com.microsoft.sqlserver.jdbc.SQLServerDriver} so that it can be loaded
+ * by reflection, as commonly expected.
+ *
+ * @author Sanne Grinovero <sanne@hibernate.org>
+ */
+public final class MsSQLReflections {
+
+    @BuildStep
+    void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        //Not strictly necessary when using Agroal, as it also registers
+        //any JDBC driver being configured explicitly through its configuration.
+        //We register it for the sake of people not using Agroal.
+        final String driverName = "com.microsoft.sqlserver.jdbc.SQLServerDriver";
```

Comment:
Can't you use the class itself to get the name? That would seem more future proof.

Reply:
Tradeoff choices.

it's not like they can change the driver name without breaking a million apps either.. I prefer to not initialize the class.